lib/conncache.c

```diff
@@ -141,7 +142,11 @@ static char *hashkey(struct connectdata *conn)
   else
     hostname = conn->host.name;
 
-  return aprintf("%s:%ld", hostname, conn->port);
+  DEBUGASSERT(len > 32);
+
+  /* put the number first so that the hostname gets cut off if too long */
+  snprintf(buf, len, "%ld%s", conn->port, hostname);
```

Comment:
Isn't snprintf C99? I believe MSVC versions prior to 2015 only have a _snprintf that doesn't append a '\0' if the string gets truncated.